### ISSUE TITLE
Object not synced if you remove and add the same file in the bucket

### ISSUE THREAD

**Issue by ron-damon** [scribd/objinsync#13]
In the following scenario, object `FOO` doesn't sync properly:  
1. :green_circle:  `FOO` object is uploaded to the bucket -> `objinsync` add them
2. :green_circle: `FOO` object is deleted from the bucket -> `objinsync` remove them
3. :red_circle: The same `FOO` object is uploaded again to the bucket -> `objinsync` doesn't add them
     If I restart `objinsync` after this step, it will sync properly.  
4. :green_circle: `FOO` object is uploaded with a modified content -> `objinsync` add them

I think it's because of the `uidCache` that is not purged when a file is deleted. 
Maybe adding a `delete` in this [`for`](https://github.com/scribd/objinsync/blob/master/pkg/sync/pull.go#L384) works:
```go
		for f, _ := range self.filesToDelete {
			os.Remove(f)
			uidKey, err := uidKeyFromLocalPath(self.LocalDir, f)
			if err == nil {
				self.uidLock.Lock()
				delete(self.uidCache, uidKey)
				self.uidLock.Unlock()
			}
		}
```

*Disclaimer: I'm not a Go expert*

**Comment by houqp** [comment]
I think you are a go expert, that is the right fix. good catch! could you send a PR?

**Comment by houqp** [comment]
This bug fix should be included in release v2.4.1
